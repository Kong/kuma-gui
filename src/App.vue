<template>
  <div id="app" class="antialiased">
    <global-header />
    <div class="flex justify-between">
      <sidebar />
      <router-view />
    </div>
  </div>
</template>

<script>
import GlobalHeader from "@/components/Global/Header";
import Sidebar from "@/components/Sidebar/Sidebar";

export default {
  components: {
    GlobalHeader,
    Sidebar
  },
  metaInfo: {
    title: "Home",
    titleTemplate: "%s | Kuma",
    htmlAttrs: {
      lang: "en"
    }
  }
};
</script>
