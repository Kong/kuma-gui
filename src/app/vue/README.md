---
section: Packages
---
# vue

Automatic `vue` initialization using our projects dependency injection layer.

