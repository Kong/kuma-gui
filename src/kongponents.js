// TODO: Typescript

import Vue from 'vue'
import '@kongponents/styles/styles.css'
import KAlert from '@kongponents/kalert'
import KBadge from '@kongponents/kbadge'
import KButton from '@kongponents/kbutton'
import KCard from '@kongponents/kcard'
import KEmptyState from '@kongponents/kemptystate'
import KIcon from '@kongponents/kicon'
import KTable from '@kongponents/ktable'
import KClipboardProvider from '@kongponents/kclipboardprovider'
import Krumbs from '@kongponents/krumbs'
import KPop from '@kongponents/kpop'
import KTabs from '@kongponents/ktabs'

Vue.component('KAlert', KAlert)
Vue.component('KBadge', KBadge)
Vue.component('KButton', KButton)
Vue.component('KCard', KCard)
Vue.component('KEmptyState', KEmptyState)
Vue.component('KIcon', KIcon)
Vue.component('KTable', KTable)
Vue.component('KClipboardProvider', KClipboardProvider)
Vue.component('Krumbs', Krumbs)
Vue.component('KPop', KPop)
Vue.component('KTabs', KTabs)
