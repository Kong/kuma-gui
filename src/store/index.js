import Vue from 'vue'
import Vuex from 'vuex'

import sidebar from '@/store/modules/sidebar'
// import workspaces from '@/store/modules/workspaces'

Vue.use(Vuex)

export default (api) => {
  const store = new Vuex.Store({
    modules: {
      sidebar
      // workspaces
    },
    state: {
      onboardingComplete: false,
      globalLoading: null,
      meshes: [],
      dataplanes: [],
      selectedMesh: 'default',
      totalMeshCount: 0,
      totalDataplaneCount: 0,
      totalDataplaneList: [],
      anyDataplanesOffline: null,
      totalDataplaneCountFromMesh: 0,
      totalTrafficRoutesCountFromMesh: 0,
      totalTrafficPermissionsCountFromMesh: 0,
      totalTrafficLogsCountFromMesh: 0,
      tagline: null,
      version: null,
      status: null
    },
    getters: {
      getOnboardingStatus (state) {
        return state.onboardingComplete
      },
      globalLoading (state) {
        return state.globalLoading
      },
      getSelectedMesh (state) {
        return state.selectedMesh
      },
      getMeshList (state) {
        return state.meshes
      },
      getDataplanes (state) {
        return state.dataplanes
      },
      getDataplanesList (state) {
        return state.totalDataplaneList
      },
      getAnyDpOffline (state) {
        return state.anyDataplanesOffline
      },
      getTotalMeshCount (state) {
        return state.totalMeshCount
      },
      getTotalDataplaneCount (state) {
        return state.totalDataplaneCount
      },
      getTotalDataplaneCountFromMesh (state) {
        return state.totalDataplaneCountFromMesh
      },
      getTotalTrafficRoutesCountFromMesh (state) {
        return state.totalTrafficRoutesCountFromMesh
      },
      getTotalTrafficPermissionsCountFromMesh (state) {
        return state.totalTrafficPermissionsCountFromMesh
      },
      getTrafficLogsFromMeshTotalCount (state) {
        return state.totalTrafficLogsCountFromMesh
      },
      getVersion (state) {
        return state.version
      },
      getTagline (state) {
        return state.tagline
      },
      getStatus (state) {
        return state.status
      }
    },
    mutations: {
      SET_ONBOARDING_STATUS (state, status) {
        state.onboardingComplete = status
      },
      SET_GLOBAL_LOADING (state, { globalLoading }) {
        state.globalLoading = globalLoading
      },
      FETCH_ALL_MESHES (state, meshes) {
        state.meshes = meshes
      },
      FETCH_DATAPLANES_FROM_MESH (state, dataplanes) {
        state.dataplanes = dataplanes
      },
      SET_SELECTED_MESH (state, mesh) {
        state.selectedMesh = mesh
      },
      SET_TOTAL_MESH_COUNT (state, count) {
        state.totalMeshCount = count
      },
      SET_TOTAL_DP_COUNT (state, count) {
        state.totalDataplaneCount = count
      },
      SET_TOTAL_DP_LIST (state, dataplanes) {
        state.totalDataplaneList = dataplanes
      },
      SET_ANY_DP_OFFLINE (state, status) {
        state.anyDataplanesOffline = status
      },
      SET_TOTAL_DP_COUNT_FROM_MESH (state, count) {
        state.totalDataplaneCountFromMesh = count
      },
      SET_TOTAL_TRAFFIC_ROUTES_COUNT_FROM_MESH (state, count) {
        state.totalTrafficRoutesCountFromMesh = count
      },
      SET_TOTAL_TRAFFIC_PERMISSIONS_COUNT_FROM_MESH (state, count) {
        state.totalTrafficPermissionsCountFromMesh = count
      },
      SET_TOTAL_TRAFFIC_LOGS_COUNT_FROM_MESH (state, count) {
        state.totalTrafficLogsCountFromMesh = count
      },
      SET_VERSION (state, version) {
        state.version = version
      },
      SET_TAGLINE (state, tagline) {
        state.tagline = tagline
      },
      SET_STATUS (state, status) {
        state.status = status
      }
    },
    actions: {
      // update the onboarding state
      updateOnboardingStatus ({ commit }, status) {
        commit('SET_ONBOARDING_STATUS', status)
      },

      // fetch all of the meshes from the API
      fetchMeshList ({ commit }) {
        return api.getAllMeshes()
          .then(response => {
            commit('FETCH_ALL_MESHES', response)
          })
          .catch(error => {
            console.error(error)
          })
      },

      // fetch all dataplanes from a specific mesh
      fetchDataplanesFromMesh ({ commit }, mesh) {
        return api.getAllDataplanesFromMesh(mesh)
          .then(response => {
            commit('FETCH_DATAPLANES_FROM_MESH', response)
          })
          .catch(error => {
            console.error(error)
          })
      },

      // update the selected mesh
      updateSelectedMesh ({ commit }, mesh) {
        commit('SET_SELECTED_MESH', mesh)
      },

      // get the total number of meshes
      getMeshTotalCount ({ commit }) {
        return api.getAllMeshes()
          .then(response => {
            const total = response.items.length

            commit('SET_TOTAL_MESH_COUNT', total)
          })
          .catch(error => {
            console.error(error)
          })
      },

      // a makeshift way to get the total amount of dataplanes present
      getDataplaneTotalCount ({ commit }) {
        const getDataplanes = async () => {
          const meshes = await api.getAllMeshes()
          const result = []

          for (let i = 0; i < meshes.items.length; i++) {
            const dataplanes = await api.getAllDataplanesFromMesh(meshes.items[i].name)
            const dpCount = await dataplanes.items.length

            result.push(dpCount)
          }

          const reduced = result.reduce((a, b) => a + b, 0)

          commit('SET_TOTAL_DP_COUNT', reduced)
        }

        getDataplanes()
      },

      // a makeshift way to get the list of all present dataplanes across all
      // meshes this will also set a status for whether or not any of the
      // dataplanes are offline
      getAllDataplanes ({ commit }) {
        const getDataplanes = async () => {
          return new Promise(async (resolve, reject) => {
            const meshes = await api.getAllMeshes()
            const result = []
            const states = []

            // create a status label column
            for (let i = 0; i < meshes.items.length; i++) {
              const mesh = meshes.items[i].name
              const dataplanes = await api.getAllDataplanesFromMesh(mesh)
              const items = await dataplanes.items

              for (let i = 0; i < items.length; i++) {
                const itemName = items[i].name
                const itemMesh = items[i].mesh

                const itemStatus = await api.getDataplaneOverviews(mesh, itemName)
                  .then(response => {
                    const items = response.dataplaneInsight.subscriptions

                    if (items && items.length > 0) {
                      for (let i = 0; i < items.length; i++) {
                        const connectTime = items[i].connectTime
                        const disconnectTime = items[i].disconnectTime

                        if (connectTime && connectTime.length && !disconnectTime) {
                          return 'Online'
                        } else {
                          return 'Offline'
                        }
                      }
                    } else {
                      return 'Offline'
                    }
                  })

                // create the full data array
                result.push({
                  status: itemStatus,
                  name: itemName,
                  mesh: itemMesh
                })
              }
            }

            // create a simple flat status object with booleans for checking
            // if any // dataplanes are offline
            for (let i = 0; i < Object.values(result).length; i++) {
              const statusVal = Object.values(result[i])[0]
              const isOnline = !(statusVal === 'Offline' || statusVal === 'offline')

              states.push(isOnline)
            }

            // if any of the dataplanes return false for being online
            // commit this so we can check against it
            const anyDpOffline = states.some(i => i === false)

            commit('SET_ANY_DP_OFFLINE', anyDpOffline)

            // commit the total list of dataplanes
            commit('SET_TOTAL_DP_LIST', result)

            // resolve the promise
            resolve()
          })
        }

        return getDataplanes()
      },

      // get the total number of dataplanes from a mesh
      getDataplaneFromMeshTotalCount ({ commit }, mesh) {
        return api.getAllDataplanesFromMesh(mesh)
          .then(response => {
            const total = response.items.length

            commit('SET_TOTAL_DP_COUNT_FROM_MESH', total)
          })
          .catch(error => {
            console.error(error)
          })
      },

      // get the total number of traffic routes from a mesh
      getTrafficRoutesFromMeshTotalCount ({ commit }, mesh) {
        return api.getTrafficRoutes(mesh)
          .then(response => {
            const total = response.items.length

            commit('SET_TOTAL_TRAFFIC_ROUTES_COUNT_FROM_MESH', total)
          })
          .catch(error => {
            console.error(error)
          })
      },

      // get the total number of traffic permissions from a mesh
      getTrafficPermissionsFromMeshTotalCount ({ commit }, mesh) {
        return api.getTrafficPermissions(mesh)
          .then(response => {
            const total = response.items.length

            commit('SET_TOTAL_TRAFFIC_PERMISSIONS_COUNT_FROM_MESH', total)
          })
          .catch(error => {
            console.error(error)
          })
      },

      // get the total number of traffic logs from a mesh
      getTrafficLogsFromMeshTotalCount ({ commit }, mesh) {
        return api.getTrafficLogs(mesh)
          .then(response => {
            const total = response.items.length

            commit('SET_TOTAL_TRAFFIC_LOGS_COUNT_FROM_MESH', total)
          })
          .catch(error => {
            console.error(error)
          })
      },

      // get the current version
      getVersion ({ commit }) {
        return api.getInfo()
          .then(response => {
            commit('SET_VERSION', response.version)
          })
          .catch(error => {
            console.error(error)
          })
      },

      // get the current tagline
      getTagline ({ commit }) {
        return api.getInfo()
          .then(response => {
            commit('SET_TAGLINE', response.tagline)
          })
          .catch(error => {
            console.error(error)
          })
      },

      // get the status of the API
      getStatus ({ commit }) {
        return api.getStatus()
          .then(response => {
            commit('SET_STATUS', response.statusText)
          })
      }
    }
  })

  return store
}
