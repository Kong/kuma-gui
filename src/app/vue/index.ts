import { createApp } from 'vue'
import {
  createRouter,
  createWebHistory,
} from 'vue-router'

import type { ServiceDefinition } from '@/services/utils'
import { token, createInjections } from '@/services/utils'
import type { Component } from 'vue'
import type { Router, RouteRecordRaw, NavigationGuard } from 'vue-router'

type Token = ReturnType<typeof token>

type VueApp = ReturnType<typeof createApp>
export type PluginDefinition = Parameters<VueApp['use']>
export type ComponentDefinition = [string, Component]

const $ = {
  app: token<(App: Component) => Promise<VueApp>>('vue.app'),
  router: token<Router>('vue.router'),

  components: token('vue.components'),
  plugins: token<PluginDefinition[]>('vue.plugins'),
  routes: token<RouteRecordRaw[]>('vue.routes'),
  routesLabel: token<RouteRecordRaw[]>('vue.routes.label'),
  navigationGuards: token<NavigationGuard[]>('vue.routes.navigation.guards'),
}
export const services = (app: Record<string, Token>): ServiceDefinition[] => {
  return [
    [$.app, {
      service: (
        components: ComponentDefinition[],
        plugins: PluginDefinition[],
      ) => {
        return async (App: Component) => {
          const app = createApp(App)

          plugins.forEach(([...args]) => {
            app.use(...args)
          })

          components.forEach(([name, component]: [string, Component]) => {
            app.component(name, component)
          })

          return app
        }
      },
      arguments: [
        $.components,
        $.plugins,
      ],
    }],
    [$.router, {
      service: (
        env: (str: 'KUMA_BASE_PATH') => string,
        routes: RouteRecordRaw[],
        guards: NavigationGuard[],
      ) => {
        const router = createRouter({
          history: createWebHistory(env('KUMA_BASE_PATH')),
          routes,
        })

        guards.forEach((item) => {
          if (typeof item === 'function') {
            router.beforeEach(item)
          }
        })

        return router
      },
      arguments: [
        app.env,
        $.routes,
        $.navigationGuards,
      ],
    }],

    [token('vue.plugins'), {
      service: (router) => {
        return [
          [router],
        ]
      },
      arguments: [
        $.router,
      ],
      labels: [
        $.plugins,
      ],
    }],

    // TODO(jc): Label decoration. Make it so we can decorate labels
    // Temporarily turn the routesLabel label into routes for decoration
    // purposes
    [$.routes, {
      service: (routes: RouteRecordRaw[]) => routes,
      arguments: [
        $.routesLabel,
      ],
    }],
    //

    // TODO(jc): Make it so we don't need to provide a [] empty default for
    // labels
    [token('application.routes.navigation.guards'), {
      service: () => {
        return []
      },
      labels: [
        $.navigationGuards,
      ],
    }],
  ]
}

export const TOKENS = $
export const [
  useRouter,
] = createInjections(
  TOKENS.router,
)
