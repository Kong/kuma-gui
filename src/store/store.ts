import { InjectionKey } from 'vue'
import { createStore, Store } from 'vuex'

import { storeConfig, State } from './index'

export const storeKey: InjectionKey<Store<State>> = Symbol('store')

export const store = createStore<State>(storeConfig)
