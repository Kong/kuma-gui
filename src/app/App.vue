<template>
  <RouteView>
    <AppLoadingBar v-if="store.state.globalLoading || route.name === undefined" />

    <template v-else>
      <AppHeader v-if="!isWizard" />

      <div v-if="route.meta.onboardingProcess">
        <router-view />
      </div>

      <div
        v-else
        class="app-content-container"
      >
        <AppSidebar v-if="!isWizard" />

        <AppView>
          <AppErrorMessage
            v-if="shouldShowAppError"
            data-testid="app-error"
          />

          <NotificationManager v-if="!isWizard && shouldShowNotificationManager" />

          <AppOnboardingNotification v-if="!isWizard && shouldShowOnboardingNotification" />

          <router-view
            :key="routeKey"
            v-slot="{ Component }"
          >
            <transition
              mode="out-in"
              name="fade"
            >
              <div
                :key="(route.name as string)"
                class="transition-root"
              >
                <component
                  :is="Component"
                  :data="props.data"
                />
              </div>
            </transition>
          </router-view>
        </AppView>
      </div>
    </template>
  </RouteView>
</template>

<script lang="ts" setup>
import { computed, watch } from 'vue'
import { useRoute } from 'vue-router'

import AppView from '@/app/application/components/app-view/AppView.vue'
import RouteView from '@/app/application/components/route-view/RouteView.vue'
import {
  useAppSidebar,
  useAppHeader,
  useAppLoadingBar,
  useAppErrorMessage,
  useNotificationManager,
  useAppOnboardingNotification,
} from '@/components'
import { useStore } from '@/store/store'
const props = defineProps({
  data: {
    type: Object,
    required: false,
    default: undefined,
  },
})
const [
  AppSidebar,
  AppHeader,
  AppLoadingBar,
  AppErrorMessage,
  NotificationManager,
  AppOnboardingNotification,
] = [
  useAppSidebar(),
  useAppHeader(),
  useAppLoadingBar(),
  useAppErrorMessage(),
  useNotificationManager(),
  useAppOnboardingNotification(),
]
const store = useStore()
const route = useRoute()

/**
 * The `router-view`’s `key` attribute value.
 *
 * Set to the current `route.path` to trigger an explicit re-render via Vue’s `key` mechanism when navigating between routes that only change in dynamic path segments while matching the same route definition.
 *
 * **Example**:
 *
 * From: /mesh/default/services/backend
 * To: /mesh/default/services/ingress
 *
 * Both routes resolve to the same route definition and the router’s default behavior is to not re-render the component in such navigations.
 */
const routeKey = computed(() => route.path)
const isWizard = computed(() => route.meta.isWizard === true)
const shouldShowAppError = computed(() => store.getters.shouldShowAppError)
const shouldShowNotificationManager = computed(() => store.getters.shouldShowNotificationManager)
const shouldShowOnboardingNotification = computed(() => store.getters.shouldShowOnboardingNotification)

watch(() => isWizard.value, setIsWizardPageClass, { immediate: true })

/**
 * Adds a class for wizard pages to the body element. This is used to control certain layout aspects of the app.
 */
function setIsWizardPageClass(isWizard: boolean) {
  const hasClass = document.body.classList.contains('is-wizard-page')

  if (isWizard && !hasClass) {
    document.body.classList.add('is-wizard-page')
  } else if (!isWizard && hasClass) {
    document.body.classList.remove('is-wizard-page')
  }
}
</script>

<style lang="scss" scoped>
body:not(.is-wizard-page) .app-content-container {
  padding-top: var(--AppHeaderHeight, initial);
  display: grid;
  grid-template-columns: var(--AppSidebarWidth) 1fr;
}

body:not(.is-wizard-page) .app-main-content {
  padding: var(--AppGap);
}
</style>
