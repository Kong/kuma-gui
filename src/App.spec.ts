import { createRouter, createWebHashHistory } from 'vue-router'
import { RouterLinkStub } from '@vue/test-utils'
import { render, screen, waitForElementToBeRemoved } from '@testing-library/vue'
import { KIcon } from '@kong/kongponents'

import { store, storeKey } from '@/store/store'
import App from './App.vue'

const router = createRouter({
  history: createWebHashHistory(),
  routes: [
    {
      path: '/',
      name: 'home',
      component: { template: 'TestComponent' },
    },
  ],
})

function renderComponent({ status = 'OK' }: { status?: string }) {
  store.state.config.status = status

  return render(App, {
    global: {
      plugins: [router, [store, storeKey]],
      stubs: {
        'router-link': RouterLinkStub,
      },
      components: {
        // TODO: Remove this once https://github.com/Kong/kongponents/pull/806 is merged and published and the library updated.
        KIcon,
      },
    },
  })
}

describe('App.vue', () => {
  it('renders main view when succesful', async () => {
    renderComponent({ status: 'OK' })

    await waitForElementToBeRemoved(() => screen.queryByRole('progressbar'))

    expect(screen.getByText('TestComponent')).toBeInTheDocument()
  })

  it('fails to renders basic view', async () => {
    const { container } = renderComponent({ status: 'ERROR' })

    await waitForElementToBeRemoved(() => screen.queryByRole('progressbar'))

    expect(container).toMatchSnapshot()
  })
})
