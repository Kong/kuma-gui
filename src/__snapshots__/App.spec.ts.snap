// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue fails to renders basic view 1`] = `
<div>
  <div
    id="app"
  >
    <div>
      <header
        class="main-header px-4 py-1 flex justify-between items-center"
      >
        <div
          class="py-1 md:py-0 flex justify-between items-center"
        >
          <a
            aria-current="page"
            class="logo router-link-exact-active router-link-active"
            href="#/"
          >
            <img
              alt="Kuma Logo"
              src="@/assets/images/product-logo.png?external"
            />
          </a>
           
          <div
            class="my-0 mx-6 upgrade-check-wrapper"
          >
            <div
              class="upgrade-check"
            >
              <!---->
            </div>
          </div>
        </div>
         
        <div
          class="flex justify-between items-center"
        >
          <div
            class="k-alert py-1 success small"
            data-v-14c8a976=""
            role="alert"
          >
            <!---->
            <!---->
            <span
              data-v-14c8a976=""
            >
              <div
                class="alert-content flex items-center text-sm"
                data-v-14c8a976=""
              >
                <div
                  class="mr-3"
                  data-v-14c8a976=""
                >
                  
        Do more with Kuma 
                  <span
                    class="alert-content__arrow"
                    data-v-14c8a976=""
                  >
                    â†’
                  </span>
                </div>
                 
                <div
                  data-v-14c8a976=""
                >
                  <a
                    class="k-button no-underline success-button small primary"
                    data-v-14c8a976=""
                    data-v-38fa8ecf=""
                    href="https://kuma.io/enterprise/?utm_source=Kuma&utm_medium=Kuma-GUI"
                    target="_blank"
                    type="button"
                  >
                    
          Enterprise
        
                  </a>
                </div>
              </div>
            </span>
          </div>
           
          <!---->
        </div>
      </header>
       
      <main
        class="main-content"
      >
        <div
          class="empty-state-wrapper global-api-status empty-state--wide-content empty-state--compact"
          data-v-567e03f2=""
        >
          <div
            class="empty-state-title"
            data-v-567e03f2=""
          >
            <!---->
            <h5
              data-v-567e03f2=""
            >
              <svg
                class="kong-icon kong-icon--centered kong-icon-warning"
                data-v-567e03f2=""
                data-v-a7c9c1d2=""
                height="64"
                role="img"
                viewBox="0 0 48 42"
                width="64"
              >
                <title
                  data-v-a7c9c1d2=""
                >
                  warning
                </title>
                <g
                  data-v-a7c9c1d2=""
                >
                  <path
                    d="M25.129733,2.64556484 C25.0166566,2.45204254 24.8546193,2.29069544 24.6593031,2.17754095 C24.0339615,1.81525556 23.2314423,2.02651215 22.8697255,2.64556484 L2.17517915,38.0628158 C2.06036961,38.2593042 2,38.4821557 2,38.7088951 C2,39.4200371 2.58244376,40 3.30518292,40 L44.6942755,40 C44.9262204,40 45.1538685,39.9388552 45.3538494,39.8229983 C45.9741835,39.4636139 46.1830669,38.6768553 45.8242793,38.0628158 L25.129733,2.64556484 Z M26.8565586,1.63656862 L47.5511049,37.0538196 C48.4690879,38.6248825 47.9342141,40.6394832 46.3564308,41.5535561 C45.8517121,41.8459599 45.2782053,42 44.6942755,42 L3.30518292,42 C1.4797808,42 3.66949018e-16,40.5265221 0,38.7088951 C0,38.1274525 0.154699044,37.5563885 0.448353557,37.0538196 L21.1428999,1.63656862 C22.0608829,0.0655056975 24.0841012,-0.467089888 25.6618845,0.446983085 C26.1568841,0.733756171 26.5685588,1.14367738 26.8565586,1.63656862 Z"
                    data-v-a7c9c1d2=""
                    fill="var(--black-75)"
                    fill-rule="nonzero"
                    id="Triangle"
                    type="secondary"
                  />
                  <path
                    d="M25.129733,2.64556484 C25.0166566,2.45204254 24.8546193,2.29069544 24.6593031,2.17754095 C24.0339615,1.81525556 23.2314423,2.02651215 22.8697255,2.64556484 L2.17517915,38.0628158 C2.06036961,38.2593042 2,38.4821557 2,38.7088951 C2,39.4200371 2.58244376,40 3.30518292,40 L44.6942755,40 C44.9262204,40 45.1538685,39.9388552 45.3538494,39.8229983 C45.9741835,39.4636139 46.1830669,38.6768553 45.8242793,38.0628158 L25.129733,2.64556484 Z"
                    data-v-a7c9c1d2=""
                    fill="var(--yellow-300)"
                    fill-rule="nonzero"
                    id="Path"
                  />
                  <path
                    d="M23,13 L25,13 C25.5522847,13 26,13.4477153 26,14 C26,16 26,18 26,20 C26,22.6594889 25.7904331,25.0570193 25.3712994,27.192591 C25.2792336,27.6616739 24.8680521,28 24.3900199,28 L23.6099609,27.9999816 C23.1319364,27.9999816 22.7207615,27.6616608 22.6286994,27.1925851 C22.2095665,25.0570148 22,22.6594864 22,20 C22,18 22,16 22,14 C22,13.4477153 22.4477153,13 23,13 Z M24,31 C25.1045695,31 26,31.8954305 26,33 C26,34.1045695 25.1045695,35 24,35 C22.8954305,35 22,34.1045695 22,33 C22,31.8954305 22.8954305,31 24,31 Z"
                    data-v-a7c9c1d2=""
                    fill="var(--black-75)"
                    id="!"
                    type="secondary"
                  />
                </g>
              </svg>
              
    Unable to reach the API
  
            </h5>
          </div>
          <div
            class="empty-state-content"
            data-v-567e03f2=""
          >
            <p
              data-v-567e03f2=""
            >
              <p
                data-v-567e03f2=""
              >
                
      Please double check to make sure it is up and running
      
                <span
                  data-v-567e03f2=""
                >
                  , and it is reachable at 
                  <code
                    data-v-567e03f2=""
                  >
                    /
                  </code>
                </span>
              </p>
            </p>
            <p
              data-v-567e03f2=""
            >
              <!---->
            </p>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
